### PR TITLE
Requestitems

### PR DESCRIPTION
**Pull request by kellysoh**
Updates: 
- You can see both open requests and requested items in the request Items Page
- You can request unlisted items on "I Need Something" Page

### CODE REVIEW

File: lib/consumer_screen/consumer_request_form.dart [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                      ),
+                    ),
+                    Spacer(),
+                    RaisedButton(
+                      onPressed: () {
+                        _getImageFromGallery();
+                      },
+                      child: Row(
+                        children: <Widget>[
+                          const Icon(Icons.phone_android),
```
Comment (judydhayoung) [inline lib/consumer_screen/consumer_request_form.dart:149]: you may go ahead comment out the gallery button

File: lib/consumer_screen/consumer_request_form.dart [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                _imageFile == null
+                    ? Text("No image selected or captured")
+                    : Center(
+                        child: Image.file(_imageFile, fit: BoxFit.cover),
+                      ),
+                Row(
+                  children: <Widget>[
+                    Spacer(),
+                    RaisedButton(
+                      onPressed: () {
```
Comment (judydhayoung) [inline lib/consumer_screen/consumer_request_form.dart:129]: May comment out the camera button

File: lib/consumer_screen/consumer_request_form.dart [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+                ),
+                ActionButton(
+                  label: 'Request',
+                  onPressed: () async {
+                    final form = _formKey.currentState;
+                    form.save();
+                    if (form.validate()) {
+                      print(this.category);
+                      // String itemID = await itemService.addItem(Item(name, description,
+                      //     category, "irrelevant", qty, "", "", qty));
```
Comment (judydhayoung) [inline lib/consumer_screen/consumer_request_form.dart:168]: please remove this dead code

File: lib/supplier_screen/order_delivery_page.dart [hunk 4]
```diff
@@ -190,15 +191,20 @@ class _OrderDeliveryPageState extends State<OrderDeliveryPage> {
...
                     widget.order.orderId, OrderStatus.Confirmed);
-
+                
                 await orderService.assignMeToOrder(
-                    widget.order.orderId, id, widget.order.itemId,
+                    widget.order.orderId, id, email, widget.order.itemId,
                     deliveryMethod:
                         DeliveryMethodUtils.convertDeliveryMethod2String(
-                            method)); //assign me to selected order as supplier
+                            method)); //assign me to selected order as supplie
```
Comment (judydhayoung) [inline lib/supplier_screen/order_delivery_page.dart:207]: supplier mis-spelled

File: lib/supplier_screen/supply_request_list.dart [hunk 5]
```diff
@@ -20,6 +20,9 @@ class _SupplyRequestListState extends State<SupplyRequestList> {
   // TODO: add state variables and methods
 
   List<Order> orders = [];
+  List<Order> selectedOrders = [] ; 
+  List<Order> openRequests = []; 
+  String dropdownValue ="My Donation List";
```
Comment (judydhayoung) [inline lib/supplier_screen/supply_request_list.dart:25]: may consider use "My Donations" instead

File: lib/supplier_screen/supply_request_list.dart [hunk 6]
```diff
@@ -35,20 +38,34 @@ class _SupplyRequestListState extends State<SupplyRequestList> {
...
 
   @override
   Widget build(BuildContext context) {
-    var userService =
-                    Provider.of<UserService>(context);
+    var userService = Provider.of<UserService>(context);
     String myId = userService.user.id;
-    readOrdersOfSupplier(myId);
     List<Order> filteredOrders;
+    readOrdersOfSupplier(myId);
```
Comment (judydhayoung) [inline lib/supplier_screen/supply_request_list.dart:58]: Please add an if statement so the these DB reads only happen once when user visit this screen.
Ask me(Isaac) for clarification.

File: lib/supplier_screen/supply_request_list.dart (hunk 5 above)
Comment (kellysoh) [reply lib/supplier_screen/supply_request_list.dart:25]: For List Name? or dropdown value?

**Pull request merged**